recipes/cctbx/meta.yaml

```diff
@@ -0,0 +1,104 @@
+{% set version = "2020.8" %}
+
+package:
+  name: cctbx-base
+  version: {{ version }}
+
+source:
+  url: https://github.com/cctbx/cctbx_project/releases/download/v{{ version }}/cctbx-{{ version }}.tar.gz
+  sha256: 38b41bb5e3b968e499a35c30de81147abb946c8de2728e90676f24843348b033
+  patches:
+    - crys3d.patch
+    - cbf.patch  # [win]
+    - libtbx.patch
+
+build:
+  number: 0
+  missing_dso_whitelist:
+    - '*/libGL.so.1'  # [linux]
+    - '*/libGLU.so.1'  # [linux]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+    - {{ cdt('mesa-libglu-devel') }}  # [linux]
+    - {{ cdt('mesa-dri-drivers') }}  # [linux]
+    - {{ cdt('libselinux') }}  # [linux]
+    - {{ cdt('libxdamage') }}  # [linux]
+    - {{ cdt('libxxf86vm') }}  # [linux]
+    - {{ cdt('libxext') }}     # [linux]
+  host:
+    - boost
+    - boost-cpp
+    - eigen
+    - future
+    - hdf5
+    - libtiff
+    - msgpack-c
+    - numpy
+    - pyopengl
+    - python
+    - python.app  # [osx]
+    - scons
+    - setuptools
+    - six
+
+outputs:
+  - name: cctbx-base
+    requirements:
+      run:
+        - biopython
+        - {{ pin_compatible('boost') }}
+        - boost-cpp
+        - cython
+        - eigen
+        - future
+        - ipython
+        - libsvm
+        - matplotlib-base
+        - mrcfile
+        - {{ pin_compatible('numpy') }}
+        - pillow
+        - psutil
+        - pyopengl
+        - python
+        - python.app  # [osx]
+        - reportlab
+        - six
+    test:
+      imports:
+        - scitbx.array_family.flex
+
+  - name: cctbx
+    build:
+    requirements:
+      run:
+        - {{ pin_subpackage('cctbx-base', max_pin="x.x.x") }}
+        - pyside2
+        - wxpython
+    test:
+      imports:
+        - scitbx.array_family.flex
+
+about:
+  home: https://github.com/cctbx/cctbx_project
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE.txt
```

Comment:
This is actually not true. Neither cctbx nor the enclosed license file are BSD-3-Clause. Granted, this is a can of worms, but worth getting it right. (Usual disclaimer: I'm not a lawyer.)

The license file contains an additional derivate works provision, making it not BSD-3.

Further, this package is vendoring other source code with different licenses that are not acknowledged. A cursory sweep found:

* boost licenses in 
  * `site-packages/boost_adaptbx/graph/breadth_first_search.py`
  * `site-packages/scitbx/auto_array.h`
  * `site-packages/scitbx/random/mersenne_twister.h`
* LGPL
  * `site-packages/ccp4io/`
  * `site-packages/cctbx/uctbx/determine_unit_cell/lgpl.txt`
  * `site-packages/libtbx/clusterTests.py`
  * `site-packages/libtbx/cluster.py`
* MIT
  * `site-packages/libtbx/objgraph.py`
* STL in
  * `site-packages/scitbx/array_family/shared_plain.h`
* custom licenses in 
  * `site-packages/chiltbx`
  * `site-packages/libtbx/fastentrypoints.py`
  * `site-packages/libtbx/xmlrpc_utils.py`
  * `site-packages/scitbx/math/parabolic_cylinder_d.h`
  * `site-packages/ucif/cifParser.h` 
  * `site-packages/wxtbx/checklistctrl.py`
  * `site-packages/wxtbx/metallicbutton.py`
  * `site-packages/wxtbx/sequence_view.py`
* "Copyright (C) Diamond Light Source", some of which are declared BSD, some are not.
  * `site-packages/cctbx/eltbx/attenuation_coefficient.cpp`
  * `site-packages/cctbx/command_line/brehm_diederichs.py`
  * `site-packages/iotbx/xds/spot_xds.py`
  * `site-packages/iotbx/xds/correction.py`
  * `site-packages/iotbx/xds/blank_cbf.py`
  * `site-packages/iotbx/xds/xds_inp.py`
  * `site-packages/iotbx/xds/gain_cbf.py`
  * `site-packages/iotbx/xds/xds_cbf.py`
  * `site-packages/iotbx/xds/xparm.py`
  * `site-packages/iotbx/xds/geometry.py`
  * `site-packages/scitbx/glmtbx/glm.h`
  * `site-packages/scitbx/glmtbx/robust_glm.h`
  * `site-packages/scitbx/glmtbx/family.h`
  * `site-packages/scitbx/glmtbx/boost_python/ext.cc`

The different licenses should at least be aggregated in a large license file on the top level, showing which licenses apply where, to comply with all the required crediting. For the purposes of the conda recipe the package as a whole then needs to be declared as mixed-licensed.

The good news is though that fundamentally everything looks like it is legal to redistribute - apart from the Diamond files that don't contain any license reference, but I have it on good authority that those are fine 🙂 (and we should fix them upstream).

Reply:
Some of these are a copyright, which is different from a license.

Also, something like `site-packages/scitbx/array_family/shared_plain.h` contain modifications allowed by the license and we follow the terms of the license in the file.
```
/* This code is derived in part from:
 *   STLport-4.5.3/stlport/stl/_vector.h
 *   STLport-4.5.3/stlport/stl/_vector.c
 *
 * Copyright (c) 1994
 * Hewlett-Packard Company
 *
 * Copyright (c) 1996,1997
 * Silicon Graphics Computer Systems, Inc.
 *
 * Copyright (c) 1997
 * Moscow Center for SPARC Technology
 *
 * Copyright (c) 1999
 * Boris Fomitchev
 *
 * This material is provided "as is", with absolutely no warranty expressed
 * or implied. Any use is at your own risk.
 *
 * Permission to use or copy this software for any purpose is hereby granted
 * without fee, provided the above notices are retained on all copies.
 * Permission to modify the code and to distribute modified code is granted,
 * provided the above notices are retained, and a notice that the code was
 * modified is included with the above copyright notice.
 *
 */
```
I'm fine with setting the license as mixed and adding the additional licenses.